src/util/selinux.c

```diff
@@ -326,6 +327,27 @@ int bus_selinux_check_send(BusSELinuxRegistry *registry,
         return 0;
 }
 
+static int bus_selinux_log(int type, const char *fmt, ...) {
+        _c_cleanup_(c_freep) char *message = NULL;
+        va_list ap;
+        int r;
+
+        va_start(ap, fmt);
+        r = vasprintf(&message, fmt, ap);
+        va_end(ap);
+        if (r < 0)
+                return r;
+
+        /* XXX: we don't have access to any context, so can't find
+         * the right UID to use, follow dbus-daemon(1) and use our
+         * own. */
+        r = util_audit_log(message, getuid());
```

Comment:
Why is this a `TODO` item? There is nothing to fix. The callback has no user-data so this should just say that we use our own uid, since selinux does not allow us to pass a callback. Or is there anything we can do about it, which does not end up being a major hack?

Reply:
We could take it up upstream (to ask them to give us a callback with userdata), not aware of any other work-around.